### ISSUE TITLE
Raspberry Pi Buster Node 14.x Installation results in node 10.x

### ISSUE THREAD

**Issue by garyns** [nodesource/distributions#1098]
Hi,

I follow these instructions for 14x, as root on a fresh installation of Buster:
```
# Using Debian, as root
curl -sL https://deb.nodesource.com/setup_current.x | bash -
apt-get install -y nodejs
```

However, on completion I get
```
node -v
v10.21.0
```

Am I doing something wrong?

Full output is below.
Thanks in advance.

```
root@raspberry:/home/pi# curl -sL https://deb.nodesource.com/setup_current.x | bash -

## Installing the NodeSource Node.js 14.x repo...


## Populating apt-get cache...

+ apt-get update
Get:1 http://security.debian.org buster/updates InRelease [65.4 kB]
Get:2 http://security.debian.org buster/updates/main amd64 Packages [218 kB]
Get:3 http://security.debian.org buster/updates/main i386 Packages [217 kB]
Get:4 http://security.debian.org buster/updates/main Translation-en [118 kB]
Get:5 http://security.debian.org buster/updates/non-free amd64 Packages [556 B]
Get:6 http://security.debian.org buster/updates/non-free i386 Packages [556 B]
Get:7 http://ftp.debian.org/debian buster InRelease [122 kB]
Get:8 http://archive.raspberrypi.org/debian buster InRelease [32.6 kB]
Get:9 http://archive.raspberrypi.org/debian buster/main i386 Packages [169 kB]
Get:10 http://ftp.debian.org/debian buster-updates InRelease [51.9 kB]
Get:11 http://ftp.debian.org/debian buster/main i386 Packages [7,864 kB]
Get:12 http://archive.raspberrypi.org/debian buster/main amd64 Packages [169 kB]
23% [11 Packages 1,094 kB/7,864 kB 14%]
Get:13 http://ftp.debian.org/debian buster/main amd64 Packages [7,906 kB]
Get:14 http://ftp.debian.org/debian buster/main Translation-en [5,968 kB]
Get:15 http://ftp.debian.org/debian buster/contrib amd64 Packages [50.5 kB]
Get:16 http://ftp.debian.org/debian buster/contrib i386 Packages [46.4 kB]
Get:17 http://ftp.debian.org/debian buster/contrib Translation-en [44.5 kB]
Get:18 http://ftp.debian.org/debian buster/non-free i386 Packages [76.2 kB]
Get:19 http://ftp.debian.org/debian buster/non-free amd64 Packages [87.7 kB]
Get:20 http://ftp.debian.org/debian buster/non-free Translation-en [88.8 kB]
Get:21 http://ftp.debian.org/debian buster-updates/main amd64 Packages [7,868 B]
Get:22 http://ftp.debian.org/debian buster-updates/main i386 Packages [7,864 B]
Get:23 http://ftp.debian.org/debian buster-updates/main Translation-en [5,672 B]
Get:24 http://ftp.debian.org/debian buster-updates/non-free i386 Packages [604 B]
Get:25 http://ftp.debian.org/debian buster-updates/non-free amd64 Packages [604 B]
Get:26 http://ftp.debian.org/debian buster-updates/non-free Translation-en [673 B]
Fetched 23.3 MB in 10s (2,325 kB/s)
Reading package lists... Done
N: Repository 'http://ftp.debian.org/debian buster InRelease' changed its 'Version' value from '10.3' to '10.5'

## Confirming "buster" is supported...

+ curl -sLf -o /dev/null 'https://deb.nodesource.com/node_14.x/dists/buster/Release'

## Adding the NodeSource signing key to your keyring...

+ curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
OK

## Creating apt sources list file for the NodeSource Node.js 14.x repo...

+ echo 'deb https://deb.nodesource.com/node_14.x buster main' > /etc/apt/sources.list.d/nodesource.list
+ echo 'deb-src https://deb.nodesource.com/node_14.x buster main' >> /etc/apt/sources.list.d/nodesource.list

## Running `apt-get update` for you...

+ apt-get update
Hit:1 http://security.debian.org buster/updates InRelease
Hit:2 http://ftp.debian.org/debian buster InRelease
Hit:3 http://archive.raspberrypi.org/debian buster InRelease
Hit:4 http://ftp.debian.org/debian buster-updates InRelease
Get:5 https://deb.nodesource.com/node_14.x buster InRelease [4,584 B]
Get:6 https://deb.nodesource.com/node_14.x buster/main amd64 Packages [764 B]
Fetched 5,348 B in 2s (2,992 B/s)
Reading package lists... Done

## Run `sudo apt-get install -y nodejs` to install Node.js 14.x and npm
## You may also need development tools to build native addons:
     sudo apt-get install gcc g++ make
## To install the Yarn package manager, run:
     curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
     echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
     sudo apt-get update && sudo apt-get install yarn


root@raspberry:/home/pi#
root@raspberry:/home/pi# apt-get install -y nodejs
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  libc-ares2 libnode64 libuv1 nodejs-doc
Suggested packages:
  npm
The following NEW packages will be installed:
  libc-ares2 libnode64 libuv1 nodejs nodejs-doc
0 upgraded, 5 newly installed, 0 to remove and 265 not upgraded.
Need to get 7,047 kB of archives.
After this operation, 30.2 MB of additional disk space will be used.
Get:1 http://ftp.debian.org/debian buster/main i386 libc-ares2 i386 1.14.0-1 [88.4 kB]
Get:2 http://ftp.debian.org/debian buster/main i386 libuv1 i386 1.24.1-1 [116 kB]
Get:3 http://ftp.debian.org/debian buster/main i386 libnode64 i386 10.21.0~dfsg-1~deb10u1 [5,783 kB]
Get:4 http://ftp.debian.org/debian buster/main i386 nodejs i386 10.21.0~dfsg-1~deb10u1 [87.1 kB]
Get:5 http://ftp.debian.org/debian buster/main i386 nodejs-doc all 10.21.0~dfsg-1~deb10u1 [973 kB]
Fetched 7,047 kB in 6s (1,207 kB/s)
Selecting previously unselected package libc-ares2:i386.
(Reading database ... 141916 files and directories currently installed.)
Preparing to unpack .../libc-ares2_1.14.0-1_i386.deb ...
Unpacking libc-ares2:i386 (1.14.0-1) ...
Selecting previously unselected package libuv1:i386.
Preparing to unpack .../libuv1_1.24.1-1_i386.deb ...
Unpacking libuv1:i386 (1.24.1-1) ...
Selecting previously unselected package libnode64:i386.
Preparing to unpack .../libnode64_10.21.0~dfsg-1~deb10u1_i386.deb ...
Unpacking libnode64:i386 (10.21.0~dfsg-1~deb10u1) ...
Selecting previously unselected package nodejs.
Preparing to unpack .../nodejs_10.21.0~dfsg-1~deb10u1_i386.deb ...
Unpacking nodejs (10.21.0~dfsg-1~deb10u1) ...
Selecting previously unselected package nodejs-doc.
Preparing to unpack .../nodejs-doc_10.21.0~dfsg-1~deb10u1_all.deb ...
Unpacking nodejs-doc (10.21.0~dfsg-1~deb10u1) ...
Setting up libc-ares2:i386 (1.14.0-1) ...
Setting up libuv1:i386 (1.24.1-1) ...
Setting up libnode64:i386 (10.21.0~dfsg-1~deb10u1) ...
Setting up nodejs-doc (10.21.0~dfsg-1~deb10u1) ...
Setting up nodejs (10.21.0~dfsg-1~deb10u1) ...
update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode
Processing triggers for libc-bin (2.28-10) ...
Processing triggers for man-db (2.8.5-2) ...
root@raspberry:/home/pi# node -v
v10.21.0
```

**Comment by igsu** [comment]
This usually happens due to the cache generated by a previous version, please run `apt-get clean all` and make sure to try the indicated version script, let us know if the problem persists.

**Comment by bartekpacia** [comment]
@igsu I encounter this issue on my Raspberry Pi 4B (newest Raspberry Pi OS)

**Comment by ezrover** [comment]
I am also encountering this with the latest Raspberry OS...

**Comment by ezrover** [comment]
by the way, there are so many wrong or confusing articles on the net. Some refer to sudo apt-get install node and some refer to sudo apt-get install nodejs. What is the difference between them and what is the correct syntax for installing the latest node V14+?

**Comment by GF-Huang** [comment]
Same issue,

![image](https://user-images.githubusercontent.com/4510984/103488335-9f2aed80-4e46-11eb-8b6c-0066656b1e08.png)

**Comment by bmonninkhof** [comment]
Same issue here, it drives me insane... 
I tried almost everything to upgrade node to v14 or v15, but everytime the result is the same. In Debian it seams impossible to upgrade NodeJS

**Comment by CarvellWakeman** [comment]
Same issue here, please consider reopening
![image](https://user-images.githubusercontent.com/10020591/104083380-ed4b4280-51f2-11eb-8951-01eaac1c9d88.png)

**Comment by bmonninkhof** [comment]
I made some notes during my endless battle with node / nodejs:
#remove all nodes / npm
sudo apt-get remove nodejs npm node
sudo apt-get purge nodejs

sudo rm -rf /usr/local/bin/npm 
sudo rm -rf /usr/local/share/man/man1/node* 
sudo rm -rf /usr/local/lib/dtrace/node.d
rm -rf ~/.npm
rm -rf ~/.node-gyp
sudo rm -rf /opt/local/bin/node
sudo rm -rf /opt/local/include/node
sudo rm -rf /opt/local/lib/node_modules
sudo rm -rf /usr/local/lib/node*
sudo rm -rf /usr/local/include/node*
sudo rm -rf /usr/local/bin/node*

sudo apt autoremove
sudo apt autoclean

# Install NVM;
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash

nvm install node
nvm use node
rm `which node`
nvm alias default node
nvm install —lts
nvm ls-remote
nvm install v14.15.4
nvm use —lts
npm config set scripts-prepend-node-path auto

The problem is the **system** nodejs!

**Comment by bmonninkhof** [comment]
root@RP-VB:/home/pi# nvm use —lts
N/A: version "—lts -> N/A" is not yet installed.

You need to run "nvm install —lts" to install it before using it.
root@RP-VB:/home/pi# nvm ls
       v14.15.4
        v15.5.1
default -> node (-> v15.5.1)
iojs -> N/A (default)
unstable -> N/A (default)
node -> stable (-> v15.5.1) (default)
stable -> 15.5 (-> v15.5.1) (default)
lts/* -> lts/fermium (-> v14.15.4)
lts/argon -> v4.9.1 (-> N/A)
lts/boron -> v6.17.1 (-> N/A)
lts/carbon -> v8.17.0 (-> N/A)
lts/dubnium -> v10.23.1 (-> N/A)
lts/erbium -> v12.20.1 (-> N/A)
lts/fermium -> v14.15.4

**Comment by akumpf** [comment]
For what it's worth, I tracked this down to a certificate issue. All of the `curl` commands use the `-s` option which makes it silent when they fail, but without that I found that I was actually getting the error message:

```
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

By adding the `-k` flag to all the curl commands in the script, **I was able to install node v14 without issue**. Seems like something just needs to be updated to ensure certs are valid.

**Comment by akumpf** [comment]
@igsu, should this be re-opened until valid certs are in place?

**Comment by bmonninkhof** [comment]
helloooooww everyone, I really don't know what happened, but I managed tot upgrade the NodeJS version pretty easy.
Did they change the curl command? I started from scratch with a fresh Raspbian build and followed the next steps. I did not test it on a fresh Rasbian Desktop version, need to test that version too.

1 # Install NVM;
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.37.2/install.sh | bash

2 # Install Homebridge
https://github.com/homebridge/homebridge/wiki/Install-Homebridge-on-Raspbian

3 # Install zigbee2mqtt
https://www.zigbee2mqtt.io/getting_started/running_zigbee2mqtt.html
